### PR TITLE
add a new page for Serving scaling & performance 

### PR DESCRIPTION
**Pull request by ReToCode**
JIRA: https://issues.redhat.com/browse/SRVKS-1115

**Changes**
- Adds new page for Serving scaling + performance based on findings of performance tests
- This page will be added in Serving area of official docs while also rewording our [existing general page ](https://access.redhat.com/documentation/en-us/red_hat_openshift_serverless/1.29/html/installing_serverless/preparing-serverless-install#about-serverless-supported-configs). Please also see the notes in the linked JIRA.

### CODE REVIEW

**Comment by ReToCode** [comment]
Gentle ping ^^

@nak3, @skonto, @rhuss

File: modules/serverless/pages/serving/scaleability-and-performance-of-serving.adoc [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+| 190Mi
+| 330Mi
+
+| net-kourier-controller (1)
+| 90Mi
+| 170Mi
+| 340Mi
+| 430Mi
+
+| istio-ingressgateway (1)
```
Comment (skonto) [inline modules/serverless/pages/serving/scaleability-and-performance-of-serving.adoc:126]: There are more stuff to consider when it comes to traffic it seems (https://www.linkedin.com/feed/update/urn:li:activity:7114044569297555456/)  :thinking: 
I think for Istio we need some pointers about its performance.

Comment (ReToCode) [reply modules/serverless/pages/serving/scaleability-and-performance-of-serving.adoc:126]: Agreed, probably need to talk to Lukas to get something from SM here. In the end its not up to us to document the specifics for SM.

Comment (skonto) [reply modules/serverless/pages/serving/scaleability-and-performance-of-serving.adoc:126]: One thing what we might check in the future is if we can optimize envoy for Kourier, but that is another topic.

File: modules/serverless/pages/serving/scaleability-and-performance-of-serving.adoc [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+            memory: <3>
+
+  podDisruptionBudgets: <4>
+    - name: name-of-pod-disruption-budget
+      minAvailable: 1
+----
+<1> Set this to at least 2, to make sure you have always at least two instances of every component running. You can also use `workloads` to override the replicas for certain components.
+<2> Use the `workloads` list to configure specific components. Use the `deployment` name of the component (like `activator`, `autoscaler`, `autoscaler-hpa`, `controller`, `webhook`, `net-kourier-controller`, `3scale-kourier-gateway`, `net-istio-controller`) and set the `replicas`.
+<3> Set the requested and limited CPU + Memory according to at least the idle consumption (see above) while also taking the above findings and your own test results into consideration.
+<4> Adjust the `PodDistruptionBudgets` to a value lower or equal to the `replicas`, to avoid problems during node maintenance.
```
Comment (nak3) [inline modules/serverless/pages/serving/scaleability-and-performance-of-serving.adoc:261]: Is it better to add a default value is `1` on all deployments?
I just got an impression that we need to take care `podDisruptionBudgets` whenever changing the `replicas`.

**Review by nak3 — COMMENTED** [review]
One comment but LGTM otherwise.

File: modules/serverless/pages/serving/scaleability-and-performance-of-serving.adoc (hunk 2 above)
Comment (ReToCode) [reply modules/serverless/pages/serving/scaleability-and-performance-of-serving.adoc:261]: I think we already have 1 as a default:
- https://github.com/openshift-knative/net-kourier/blob/release-v1.11/config/300-gateway.yaml#L205
- https://github.com/openshift-knative/serving/blob/release-v1.11/config/core/deployments/webhook-hpa.yaml#L51

Comment (nak3) [reply modules/serverless/pages/serving/scaleability-and-performance-of-serving.adoc:261]: Yes, so should we mention that the default is `1` in this docs? Sorry for my unclear comment.

Comment (ReToCode) [reply modules/serverless/pages/serving/scaleability-and-performance-of-serving.adoc:261]: Ah, yes we can do that 👍

Comment (ReToCode) [reply modules/serverless/pages/serving/scaleability-and-performance-of-serving.adoc]: [Added](https://github.com/openshift-knative/docs/pull/98/commits/9e62b1cf9c10d77959867c5e8e636a4529020fc6)

**Comment by rhuss** [comment]
@ReToCode thanks a lot for this very needed documentation.

Overall it looks great, the only thing that I would add is the context under which the results where obtained: OpenShift version, cluster size, OpenShift Serverless version ..., otherwise this could be misleading when future versions after a different characteristics.

I haven't proofread for typos, because I want first the opinion of the docs team about the structure and wording. In the paster, there where some strong opinions how things should be structured in sections (to conform to the overall, very strict OpenShift style-guide), I'm not sure if this is still valid. Nevertheless, I'm happy to do a final round after integration into the regular docs (I really would add them not at a "Dev Prev" feature).

**Comment by ReToCode** [comment]
> Overall it looks great, the only thing that I would add is the context under which the results where obtained: OpenShift version, cluster size, OpenShift Serverless version ..., otherwise this could be misleading when future versions after a different characteristics.

Added. 

> I haven't proofread for typos, because I want first the opinion of the docs team about the structure and wording. In the paster, there where some strong opinions how things should be structured in sections (to conform to the overall, very strict OpenShift style-guide), I'm not sure if this is still valid. Nevertheless, I'm happy to do a final round after integration into the regular docs (I really would add them not at a "Dev Prev" feature).

That's fine. The idea here is that the docs team can take from that write-up what they need and can reword/restructure as they like.

**Pull request merged**